api/client/proxy/proxy_test.go

```diff
@@ -111,12 +117,46 @@ func TestGetProxyAddress(t *testing.T) {
 			for _, env := range tt.env {
 				t.Setenv(env.name, env.val)
 			}
-			p := GetProxyAddress(tt.targetAddr)
+			p := GetProxyURL(tt.targetAddr)
 			if tt.proxyAddr == "" {
 				require.Nil(t, p)
 			} else {
 				require.NotNil(t, p)
 				require.Equal(t, tt.proxyAddr, p.Host)
+				require.Nil(t, p.User)
+			}
+
+			// now reconstruct the env var with user:pass@host, set the env again, and test parsing that
+			expectedUser := "alice"
+			expectedPassword := "password"
+			for _, env := range tt.env {
+				if strings.EqualFold(env.name, "http_proxy") || strings.EqualFold(env.name, "https_proxy") {
+					var val string
+					if tt.isRawProxyAddr {
+						// url.Parse will parse a raw host:port as scheme:opaque,
+						// so instead just prepend the user info to the raw host:port proxy addr
+						val = fmt.Sprintf("%v:%v@%v", expectedUser, expectedPassword, env.val)
+					} else {
+						u, _ := url.Parse(env.val)
+						u.User = url.UserPassword(expectedUser, expectedPassword)
+						val = u.String()
+					}
+					// test with user:pass@ injected into addr
+					t.Setenv(env.name, val)
+				} else {
+					t.Setenv(env.name, env.val)
+				}
+				p := GetProxyURL(tt.targetAddr)
+				if tt.proxyAddr == "" {
+					require.Nil(t, p)
+				} else {
+					require.NotNil(t, p)
+					require.Equal(t, tt.proxyAddr, p.Host)
+					require.NotNil(t, p.User)
+					require.Equal(t, expectedUser, p.User.Username())
+					password, _ := p.User.Password()
+					require.Equal(t, expectedPassword, password)
+				}
```

Comment:
I refactored this. I think latest is much easier to reason about. I want to keep the auth test "multiplied" with the existing tests so that each scenario is tested thoroughly.